src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -296,23 +574,84 @@ inline void Details::DocumentStatisticsTrainingOnlyPolicy::fit(InputType const &
 }
 
 inline DocumentStatisticsAnnotationData Details::DocumentStatisticsTrainingOnlyPolicy::complete_training(void) {
-    IndexMap                                termIndex(Microsoft::Featurizer::Featurizers::Components::CreateIndexMap<std::string>(_termFrequency, IndexMap()));
-    return DocumentStatisticsAnnotationData(std::move(_termFrequency), std::move(termIndex), std::move(_totalNumDocuments));
+
+    //prune map by maxDf, minDf and topKTerms
+    FrequencyMap                            prunedTermFreq(PruneTermFreqMap(std::move(_termFrequency), _minDf, _maxDf, static_cast<std::float_t>(_totalNumDocuments), _topKTerms));
+
+    //In general, the custom vocabulary should assign values to words that no bigger than the map's size
+    //however, some may choose to use unique values. We do not check if the given vocabulary is appropriate
+    //But suggest customer use reasonable mapping
+    IndexMap                                termIndex(Microsoft::Featurizer::Featurizers::Components::CreateIndexMap<std::string>(prunedTermFreq,
+                                                                                                                                  _existingVocabulary.has_value() ? *_existingVocabulary : IndexMap()));
+
+    //here the termIndex includes(strictly) termFrequency, which means termIndex has all the keys that appear in termFrequency, but may has some keys not in termFrequency
+    //ignore the additional keys in termIndex
+    FrequencyAndIndexMap                    termFrequencyAndIndex(MergeTwoMapsWithSameKeys(prunedTermFreq, termIndex));
+
+
+   // BaseType::add_annotation(std::make_shared<AnnotationImpl>(EstimatorPolicyT::complete_training()), _colIndex);
+
+    return DocumentStatisticsAnnotationData(std::move(termFrequencyAndIndex), std::move(_totalNumDocuments));
 }
 
+
 // ----------------------------------------------------------------------
 // ----------------------------------------------------------------------
 // ----------------------------------------------------------------------
 template <typename SetT, typename CreateKeyFuncT, typename KeyToStringFuncT>
 void Details::DocumentStatisticsTrainingOnlyPolicy::fit_impl(InputType const &input, CreateKeyFuncT const &createKeyFunc, KeyToStringFuncT const &keyToStringFunc) {
     SetT                                    documents;
 
-    split_temp(
-        input,
-        [&createKeyFunc, &documents](InputTypeConstIterator begin, InputTypeConstIterator end) {
-            documents.insert(createKeyFunc(begin, end));
+    if (_analyzer == AnalyzerMethod::Word) {
```

Comment:
I have the same comment as before - this logic can be invoked once at class construction time rather than during each call to fit.

Reply:
yes -I haven't marked this as resolved, will soon